### PR TITLE
Harsh/fix 404 page for undefined events

### PR DESCRIPTION
**Pull request by harshsinghatz**
## What does this PR do?
- Currently 404 page is not show to the user, if the event type doesn't exist

Fixes # (issue)
- [Issue](https://github.com/calcom/cal.com/issues/5665)

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

Before:
image
After:
image

## How should this be tested?

- [ ] Try searching for a event that doesn't exist- It should show 404 
- [ ] Try searching for a valid event and it should work fine

### CODE REVIEW

**Comment by harshsinghatz** [comment]
RCA for this: [] returns true so, we have to check the actual length of the array

**Review by PeerRich — APPROVED** [review]
LGTM, waiting for another review

File: apps/web/pages/[user]/[type].tsx
Comment (hariombalhara) [inline apps/web/pages/[user]/[type].tsx:124]: nit
```suggestion
  if (!user || !user.eventTypes.length) return { notFound: true };
```

**Review by hariombalhara — APPROVED** [review]

**Comment by harshsinghatz** [comment]
@PeerRich can we merge this PR?

**Comment by PeerRich** [comment]
yes!

**Pull request merged**
